api/v1alpha2/awsmachine_conversion.go

```diff
@@ -70,9 +73,23 @@ func restoreAWSMachineSpec(restored, dst *infrav1alpha3.AWSMachineSpec) {
 			dst.NonRootVolumes = append(dst.NonRootVolumes, volume.DeepCopy())
 		}
 	}
+	if src.RootDeviceSize != 0 {
+		if dst.RootVolume == nil {
+			dst.RootVolume = &infrav1alpha3.Volume{
+				Size: src.RootDeviceSize,
+			}
+		} else {
+			dst.RootVolume.Size = src.RootDeviceSize
+		}
+	}
 
 	dst.Tenancy = restored.Tenancy
-	dst.CloudInit.SecureSecretsBackend = restored.CloudInit.SecureSecretsBackend
+
+	if restored.CloudInit.SecureSecretsBackend != "" {
+		if src.CloudInit != nil {
+			dst.CloudInit.SecureSecretsBackend = restored.CloudInit.SecureSecretsBackend
+		}
```

Comment:
Back when I added the secure secrets stuff, I had originally also added 
![image](https://user-images.githubusercontent.com/1441070/97475907-b7adf180-1945-11eb-829d-63ed08a14fc9.png) before removing it thinking it was excessive.

Do we in fact need to conditionally restore the SecretARN here too?

Reply:
SecretARN turned into SecretPrefix + SecretCount.

Reply:
And SecretPrefix + SecretCount exist in both a2 and a3